### PR TITLE
Add option -o outfile

### PR DESCRIPTION
**Pull request by yashi**
This PR is to add a new option `-o outfile`.

While adding it, I found that meson is generating a warning about run_command.  This PR includes a small fix for that as well.

### CODE REVIEW

**Comment by jpmens** [comment]
Thank you for your submission!

Can you tell us why you feel this option is necessary instead of, say, using output redirection?

**Comment by yashi** [comment]
Hi,

Thank you for a nice tool!

As I wrote in a commit message some tools exec without shell, such as CMake in my case.  without any shell, you can't use redirection, '>', to create a file.
https://cmake.org/cmake/help/latest/command/execute_process.html#execute-process

**Comment by jpmens** [comment]
I missed the commit message, sorry, and thank you for the explanation.

**Comment by jpmens** [comment]
Merged, thanks!

File: meson.build [hunk 1]
```diff
@@ -56,7 +56,8 @@ jo1 = custom_target('jo.1',
                     command: [pandoc_commands, '@OUTPUT@', '@INPUT@']).full_path()
 run_command(pandoc_commands,
             join_paths(meson.current_build_dir(), 'jo.1'),
-            join_paths(meson.current_source_dir(), 'jo.pandoc'))
+            join_paths(meson.current_source_dir(), 'jo.pandoc'),
+            check: false)
```
Comment (eli-schwartz) [inline meson.build:60]: The point of Meson adding the warning was specifically to make people consider whether they need to check the return value, so choosing "false" based purely on it being the old default seems wrong.

Let's instead look at the command in question and ponder: do we need to check this?

What happens if the pandoc command totally fails? Should this be an error? Probably.

**Pull request merged**